core/src/test/kotlin/org/evomaster/core/search/algorithms/CroAlgorithmTest.kt

```diff
@@ -0,0 +1,398 @@
+package org.evomaster.core.search.algorithms
+
+import com.google.inject.Injector
+import com.google.inject.Key
+import com.google.inject.Module
+import com.google.inject.TypeLiteral
+import com.netflix.governator.guice.LifecycleInjector
+import org.evomaster.core.BaseModule
+import org.evomaster.core.EMConfig
+import org.evomaster.core.TestUtils
+import org.evomaster.core.search.algorithms.onemax.OneMaxIndividual
+import org.evomaster.core.search.algorithms.onemax.OneMaxModule
+import org.evomaster.core.search.algorithms.onemax.OneMaxSampler
+import org.evomaster.core.search.service.ExecutionPhaseController
+import org.evomaster.core.search.service.Randomness
+import org.evomaster.core.search.algorithms.wts.WtsEvalIndividual
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+
+class CroAlgorithmTest {
+
+    private lateinit var injector: Injector
+
+    @BeforeEach
+    fun setUp() {
+        injector = LifecycleInjector.builder()
+            .withModules(* arrayOf<Module>(OneMaxModule(), BaseModule()))
+            .build().createInjector()
+    }
+
+    // Verifies that CRO can find the optimal solution for the OneMax problem
+    @Test
+    fun testCroAlgorithmFindsOptimum() {
+        TestUtils.handleFlaky {
+            val cro = injector.getInstance(
+                Key.get(object : TypeLiteral<CroAlgorithm<OneMaxIndividual>>() {})
+            )
+
+            val config = injector.getInstance(EMConfig::class.java)
+            injector.getInstance(Randomness::class.java).updateSeed(42)
+
+            // Keep defaults close to other GA tests
+            config.maxEvaluations = 10000
+            config.stoppingCriterion = EMConfig.StoppingCriterion.ACTION_EVALUATIONS
+
+            val epc = injector.getInstance(ExecutionPhaseController::class.java)
+            epc.startSearch()
+            val solution = cro.search()
+            epc.finishSearch()
+
+            assertTrue(solution.individuals.size == 1)
+            assertEquals(OneMaxSampler.DEFAULT_N.toDouble(), solution.overall.computeFitnessScore(), 0.001)
+        }
+    }
+
+    @Test
+    fun testUniMolecular_DecompositionPath_usesDecomposition() {
+        TestUtils.handleFlaky {
+            val cro = injector.getInstance(
+                Key.get(object : TypeLiteral<CroAlgorithm<OneMaxIndividual>>() {})
+            )
+
+            val config = injector.getInstance(EMConfig::class.java)
+            val randomness = injector.getInstance(Randomness::class.java)
+            randomness.updateSeed(123L)
+
+            // Single molecule guarantees uni-molecular branch regardless of collision rate
+            config.populationSize = 1
```

Comment:
setting up config like this is usually fine, as long as the settings or not used/needed in any `@PostConstruct` block